src/Graph/GraphPrinter.php

```diff
@@ -133,79 +146,102 @@ public function getDependencyError() {
 	 * @return string
 	 */
 	protected function getResultText( SMWQueryResult $res, $outputmode ) {
-		global $wgVersion;
+		// Remove this once SRF requires 3.1+
+		if ( $this->hasMissingDependency() ) {
+			return $this->getDependencyError();
+		}
 
 		// iterate query result and create SRF\GraphNodes
 		while ( $row = $res->getNext() ) {
 			$this->processResultRow( $row );
 		}
 
-		// use GraphFormatter to build the graph
+		// use GraphFormater to build the graph
 		$graphFormatter = new GraphFormatter( $this->options );
 		$graphFormatter->buildGraph( $this->nodes );
 
 		// GraphViz is not working for version >= 1.33, so we need to use the Diagrams extension
 		// and formatting is a little different from the GraphViz extension
+		global $wgVersion;
 		if ( version_compare( $wgVersion, '1.33', '>=' ) &&
 			\ExtensionRegistry::getInstance()->isLoaded( 'Diagrams' ) ) {
 			$result = "<graphviz>{$graphFormatter->getGraph()}</graphviz>";
 		} else {
-			// Calls graphvizParserHook function from MediaWiki GraphViz extension
+			// Calls graphvizParserHook function from MediaWiki GraphViz or External Data extension
```

Comment:
Shouldn't this be Graphviz, Diagrams or External Data?

Reply:
Corrected the comments.